solutions/Fully configurable/variables.tf

```
@@ -0,0 +1,316 @@
+}
+variable "existing_resource_group_name" {
+  type        = string
+  description = "Name of the existing resource group."
+}
+
+variable "region" {
+  type        = string
+  description = "The region to provision resources to."
+  default     = "us-south"
+}
+
+variable "prefix" {
+  type        = string
+  description = "The prefix to apply to all resources created by this solution."
+  default     = null
+}
+
+########################################################################################################################
+# Secrets Manager
+########################################################################################################################
+
+variable "secrets_manager_instance_name" {
+  type        = string
+  description = "The name to give the Secrets Manager instance provisioned by this solution. If a prefix input variable is specified, it is added to the value in the `<prefix>-value` format."
+  default     = "base-security-services-sm"
+}
+
+variable "existing_secrets_manager_crn" {
+  type        = string
+  description = "The CRN of an existing Secrets Manager instance. If not supplied, a new instance is created."
+  default     = null
+}
+
+variable "existing_secrets_endpoint_type" {
+  type        = string
+  description = "The endpoint type to use if existing_secrets_manager_crn is specified. Possible values: public, private."
+  default     = "private"
+  validation {
+    condition     = contains(["public", "private"], var.existing_secrets_endpoint_type)
+    error_message = "Only \"public\" and \"private\" are allowed values for 'existing_secrets_endpoint_type'."
+  }
+}
+
+variable "service_plan" {
+  type        = string
+  description = "The pricing plan to use when provisioning a Secrets Manager instance. Possible values: `standard`, `trial`. Applies only if `provision_sm_instance` is set to `true`."
+  default     = "standard"
+  validation {
+    condition     = contains(["standard", "trial"], var.service_plan)
+    error_message = "Only \"standard\" and \"trial\" are allowed values for sm_service_plan."
+  }
+}
+
+variable "allowed_network" {
+  type        = string
+  description = "The types of service endpoints to set on the Secrets Manager instance. Possible values: `private-only`, `public-and-private`."
+  default     = "private-only"
+  validation {
+    condition     = contains(["private-only", "public-and-private"], var.allowed_network)
+    error_message = "The specified allowed_network is not a valid selection."
+  }
+}
+
+variable "kms_encryption_enabled" {
+  type        = bool
+  description = "Set this to true to control the encryption keys used to encrypt the data that you store in Secrets Manager. If set to false, the data that you store is encrypted at rest by using envelope encryption. For more details, see https://cloud.ibm.com/docs/secrets-manager?topic=secrets-manager-mng-data&interface=ui#about-encryption."
+  default     = false
+}
+
+variable "secret_manager_tags" {
```

Access tag support will be coming in https://github.com/terraform-ibm-modules/terraform-ibm-secrets-manager/issues/294 so suggest to explicitly say its a resource tag:
```suggestion
variable "secret_manager_resource_tags" {
```